qutebrowser/browser/commands.py

```diff
@@ -643,14 +643,32 @@ def scroll_perc(self, perc: {'type': float}=None,
 
     @cmdutils.register(instance='command-dispatcher', hide=True,
                        scope='window', count='count')
-    def scroll_page(self, x: {'type': float}, y: {'type': float}, count=1):
+    def scroll_page(self, x: {'type': float}, y: {'type': float}, *,
+                    top_navigate: {'type': ('prev', 'decrement')}=None,
+                    bottom_navigate: {'type': ('next', 'increment')}=None,
+                    count=1):
         """Scroll the frame page-wise.
 
         Args:
             x: How many pages to scroll to the right.
             y: How many pages to scroll down.
+            bottom_navigate: :navigate to the next page on bottom
+            top_navigate: :navigate to the previous page on top
             count: multiplier
         """
+        frame = self._current_widget().page().currentFrame()
+        if not frame.url().isValid():  # Issue 701
+            return
```

Comment:
Hmm, then maybe it'd be better to check for `frame.scrollBarMaximum(Qt.Vertical) == 0` in the `if` before calling `self.navigate`? Or will this always be the case for pages without any scrollbar?

I don't know if there are any pages where the URL would always be considered invalid, and then scrolling should definitely work.